src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -50,15 +54,27 @@ pub struct WorkUnit {
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub struct StartedWorkUnit {
   pub name: String,
-  pub start_time: std::time::SystemTime,
-  pub span_id: String,
+  pub start_time: SystemTime,
+  pub span_id: SpanId,
   pub parent_id: Option<String>,
   pub metadata: WorkunitMetadata,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Hash, Default)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub struct WorkunitMetadata {
   pub desc: Option<String>,
+  pub display: bool,
+  pub running: bool,
```

Comment:
"running" doesn't feel like a boolean property of a workunit: a `StartedWorkUnit` is running, and a `WorkUnit` is finished, but they should not each have the boolean property.

Reply:
I'm going to changed this to "blocked" I think, since the point of this is for a workunit to report whether it's blocked on something. I agree that this doesn't make sense for a finished `WorkUnit` and in a follow-up commit I'd probably want to modify the types to reflect this.